ios/RNCAsyncStorage.m

```diff
@@ -385,18 +426,23 @@ - (instancetype)init
         return nil;
     }
 
-    // First migrate our deprecated path "Documents/.../RNCAsyncLocalStorage_V1" to
-    // "Documents/.../RCTAsyncLocalStorage_V1"
-    RCTStorageDirectoryMigrationCheck(
-        RCTCreateStorageDirectoryPath_deprecated(RCTOldStorageDirectory),
-        RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory),
-        YES);
+    // Get the path to any old storage directory that needs to be migrated. If multiple exist,
+    // the oldest are removed and the most recently modified is returned.
+    NSString *oldStoragePath = RCTGetStoragePathForMigration();
+    if (oldStoragePath != nil) {
+        // First migrate our deprecated path "Documents/.../RNCAsyncLocalStorage_V1" or
+        // "Documents/.../RCTsyncLocalStorage" to "Documents/.../RCTAsyncLocalStorage_V1"
+        RCTStorageDirectoryMigrationCheck(
+            oldStoragePath,
+            RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory),
+            YES);
 
-    // Then migrate what's in "Documents/.../RCTAsyncLocalStorage_V1" to "Application
-    // Support/[bundleID]/RCTAsyncLocalStorage_V1"
-    RCTStorageDirectoryMigrationCheck(RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory),
-                                      RCTCreateStorageDirectoryPath(RCTStorageDirectory),
-                                      NO);
+        // Then migrate what's in "Documents/.../RCTAsyncLocalStorage_V1" to "Application
+        // Support/[bundleID]/RCTAsyncLocalStorage_V1"
+        RCTStorageDirectoryMigrationCheck(RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory),
+                                          RCTCreateStorageDirectoryPath(RCTStorageDirectory),
+                                          NO);
```

Comment:
This migration needs to be outside of the if statement - it's not related to Expo, but normal RN projects. If someone upgrades from pre 1.8.0 version, the migration to Application Support won't kick in

Reply:
addressed this feedback, did manual testing again and works as expected